src/app/wallets/ln-send-payment.ts

```diff
@@ -306,35 +306,32 @@ const executePaymentViaIntraledger = async ({
   const recipientWallet = await WalletsRepository().findById(recipientWalletId)
   if (recipientWallet instanceof Error) return recipientWallet
 
-  const lnFee = toSats(0)
-  const sats = toSats(amount + lnFee)
-  const usd = sats * usdPerSat
-  const usdFee = lnFee * usdPerSat
+  const amountDisplayCurrency = DisplayCurrencyConversionRate(usdPerSat)(amount)
 
   return LockService().lockWalletId(
     { walletId: senderWallet.id, logger },
     async (lock) => {
       const balance = await LedgerService().getWalletBalance(senderWallet.id)
       if (balance instanceof Error) return balance
-      if (balance < sats) {
+      if (balance < amount) {
         return new InsufficientBalanceError(
-          `Payment amount '${sats}' exceeds balance '${balance}'`,
+          `Payment amount '${amount}' exceeds balance '${balance}'`,
         )
       }
 
       const journal = await LockService().extendLock({ logger, lock }, async () =>
-        LedgerService().addLnIntraledgerTxSend({
+        LedgerService().addLnIntraledgerTxTransfer({
           senderWalletId: senderWallet.id,
+          senderWalletCurrency: senderWallet.currency,
+          senderUsername: senderAccount.username,
           paymentHash,
           description,
-          sats,
-          fee: lnFee,
-          usd,
-          usdFee,
+          sats: amount,
+          amountDisplayCurrency,
```

Comment:
correct. but this will comes in another PR. currently the assumption is that wallets are still all BTC wallets. I'm working on the plumbing to send the currency up to the ledger right now.